### PR TITLE
JAMES-2334 Basic RabbitMQ cluster tests

### PR DESCRIPTION
**Pull request by aduprat**
Only two last commits:

Includes:
 - Starting a cluster
 - Production and consumption across the cluster
 - stopping nodes when consuming, producing
 - Connecting a cluster with a stopped node
 - Declaring queue and exchange on different nodes

### CODE REVIEW

**Comment by aduprat** [comment]
test this please

File: server/queue/queue-rabbitmq/pom.xml [hunk 1]
```diff
@@ -70,6 +70,15 @@
             <artifactId>testcontainers</artifactId>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>org.slf4j</groupId>
+            <artifactId>slf4j-api</artifactId>
+        </dependency>
+        <dependency>
```
Comment (chibenwa) [inline server/queue/queue-rabbitmq/pom.xml]: I did this to code fast, but maybe we should rely on logback.

File: server/queue/queue-rabbitmq/src/test/java/org/apache/james/queue/rabbitmq/DockerClusterRabbitMQExtention.java [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+        String cookie = DigestUtils.sha1Hex("secret cookie here");
+
+        String netId = DockerClientFactory.instance().client().createNetworkCmd()
+            .withName("rabbitCluster2")
+            .withCheckDuplicate(true)
+            .withEnableIpv6(false)
+            .exec()
+            .getId();
+
+        network = new Network() {
```
Comment (chibenwa) [inline server/queue/queue-rabbitmq/src/test/java/org/apache/james/queue/rabbitmq/DockerClusterRabbitMQExtention.java]: That's very hacky....

I'd propose a refined version:

```
        network = Network.NetworkImpl.builder()
            .enableIpv6(false)
            .createNetworkCmdModifiers(ImmutableList.of())
            .build();
```

File: server/queue/queue-rabbitmq/src/test/java/org/apache/james/queue/rabbitmq/RabbitMQClusterTest.java [hunk 3]
```diff
@@ -0,0 +1,270 @@
...
+        }
+
+        assertThat(result)
+            .containsOnlyElementsOf(
+                IntStream.range(0, MESSAGE_COUNT)
+                    .boxed()
+                    .collect(Guavate.toImmutableList()));
+    }
+
+    private DefaultConsumer callback(ConcurrentLinkedDeque<Integer> result, Channel channel2) {
```
Comment (chibenwa) [inline server/queue/queue-rabbitmq/src/test/java/org/apache/james/queue/rabbitmq/RabbitMQClusterTest.java]: Why can't we use your awesome inmemoryConsumer stuff, that does the exact same job.

Comment (chibenwa) [reply server/queue/queue-rabbitmq/src/test/java/org/apache/james/queue/rabbitmq/RabbitMQClusterTest.java]: s/channel2/channel/

File: server/queue/queue-rabbitmq/src/test/java/org/apache/james/queue/rabbitmq/RabbitMQClusterTest.java [hunk 4]
```diff
@@ -0,0 +1,270 @@
...
+        return new DefaultConsumer(channel2) {
+            @Override
+            public void handleDelivery(String consumerTag, Envelope envelope, AMQP.BasicProperties properties, byte[] body) {
+
```
Comment (chibenwa) [inline server/queue/queue-rabbitmq/src/test/java/org/apache/james/queue/rabbitmq/RabbitMQClusterTest.java]: Extra line

File: server/queue/queue-rabbitmq/src/test/java/org/apache/james/queue/rabbitmq/RabbitMQClusterTest.java [hunk 5]
```diff
@@ -0,0 +1,270 @@
...
+                result.add(Integer.valueOf(new String(body, StandardCharsets.UTF_8)));
+            }
+        };
+    }
+
+    private DefaultConsumer callbackWithCrash(ConcurrentLinkedDeque<Integer> result, Channel channel2, AtomicInteger atomicInteger, DockerRabbitMQ containerToStop) {
+        return new DefaultConsumer(channel2) {
+            @Override
+            public void handleDelivery(String consumerTag, Envelope envelope, AMQP.BasicProperties properties, byte[] body) {
+                if (atomicInteger.incrementAndGet() == 5) {
```
Comment (chibenwa) [inline server/queue/queue-rabbitmq/src/test/java/org/apache/james/queue/rabbitmq/RabbitMQClusterTest.java]: We could add an optional operation to the inMemoryConsumer to handle this...

File: server/queue/queue-rabbitmq/src/test/java/org/apache/james/queue/rabbitmq/RabbitMQClusterTest.java [hunk 6]
```diff
@@ -0,0 +1,270 @@
...
+    }
+
+    @Test
+    public void nodeKillingWhenConsuming(DockerRabbitMQCluster cluster) throws Exception {
+        ConcurrentLinkedDeque<Integer> result = new ConcurrentLinkedDeque<>();
+
+        ConnectionFactory connectionFactory1 = RabbitMQUtils.prepareConnectionFactory(cluster.getRabbitMQ1());
+        ConnectionFactory connectionFactory2 = RabbitMQUtils.prepareConnectionFactory(cluster.getRabbitMQ2());
+
+        ImmutableList<Address> addresses = ImmutableList.of(
```
Comment (chibenwa) [inline server/queue/queue-rabbitmq/src/test/java/org/apache/james/queue/rabbitmq/RabbitMQClusterTest.java]: Could this be returned by the cluster? `cluster.getNodeAddresses`

File: server/queue/queue-rabbitmq/pom.xml (hunk 1 above)
Comment (mbaechler) [reply server/queue/queue-rabbitmq/pom.xml]: of course

**Comment by mbaechler** [comment]
merged in another PR

**Pull request closed without merging**